Spigot-Server-Patches/0336-Extend-player-profile-API-to-support-skin-changes.patch

```diff
@@ -0,0 +1,64 @@
+From 95e357f7dd549b278714a9ea76bf6d11d6c2b5fd Mon Sep 17 00:00:00 2001
+From: NickAcPT <32451103+NickAcPT@users.noreply.github.com>
+Date: Sat, 21 Jul 2018 01:30:30 +0100
+Subject: [PATCH] Extend player profile API to support skin changes
+
+Added code that refreshes the player's skin by sending packets with a special order, telling the client to respawn the player and re-apply the game profile
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 6cbf429f..57f17120 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -55,6 +55,7 @@ import org.bukkit.craftbukkit.CraftStatistic;
+ import org.bukkit.craftbukkit.CraftWorld;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancement;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancementProgress;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.craftbukkit.map.CraftMapView;
+ import org.bukkit.craftbukkit.map.RenderData;
+ import org.bukkit.craftbukkit.scoreboard.CraftScoreboard;
+@@ -1176,10 +1177,41 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         for (EntityPlayer player : players) {
+             player.getBukkitEntity().reregisterPlayer(self);
+         }
++        refreshPlayer();
+     }
+     public PlayerProfile getPlayerProfile() {
+         return new CraftPlayerProfile(this).clone();
+     }
++
++    private void refreshPlayer() {
++        EntityPlayer entityplayer = getHandle();
++
++        Packet respawn = new PacketPlayOutRespawn(entityplayer.dimension, entityplayer.world.getDifficulty(), entityplayer.world.getWorldData().getType(), entityplayer.playerInteractManager.getGameMode());
++        Location l = getLocation();
++        Packet pos = new PacketPlayOutPosition(l.getX(), l.getY(), l.getZ(), l.getYaw(), l.getPitch(), new HashSet<>(), 0);
```

Comment:
Ok going to bunch everything up into one review comment, overall this is looking a lot better now.

But lets clean up some naming/impl details.

1) remove the import up top for CraftItemStack, remnant from previous version
2) rename l to loc, we prefer to not have 1 letter variables outside of positional stuff like x/y/z
3) rename playerCon to just connection
4) inline the packets so its connection.sendPacket(new PacketPlayOut*), we don't need variables for that too.
5) Add a comment on where this logic can be found in EntityPlayer or such incase future updates change the logic and we need to update this.

I think that'll cover it on quality wise and be good to go at that point.